src/routes/App.tsx

```diff
@@ -7,8 +8,33 @@ import BookLocation from '../screens/BookLocation'
 import Register from '../screens/Register'
 import ManageProfile from '../screens/ManageProfile'
 import ManageLocation from '../screens/provider/ManageLocation'
+import { UserInterface } from '../interfaces/user.interfaces'
+
+const URL = import.meta.env.VITE_API_URL
 
 function App() {
+  const [user, setUser] = useState<UserInterface | null>(null)
+
+  useEffect(() => {
+    // fetch to get user info
+    // if user is logged in, set user info in context
+    async function getUserInfo() {
+      const response = await fetch(`${URL}/users/me`, {
+        headers: { 'Content-Type': 'application/json' },
+        credentials: 'include',
+      })
+
+      const data = await response.json()
+
+      if (response.ok) {
+        setUser(data)
+        console.log(data)
+      }
+    }
+
+
+  },[])
```

Comment:
I think when we use useEffect hook it will fetch data twice Let me see how to fix the problem first

Reply:
![image](https://user-images.githubusercontent.com/88868804/221232228-a98c275d-1506-4e1a-8585-b7e40d0db5bb.png)
After doing some research,I found that useEffect in initial render is OK. then we can ignore about twice in render.
But you need to fix some point
1. In useEffect need to be method like this that we will understand what going on in initial render phase
2. I think maybe we call it fetchUser is more meaningful that we will understand what this function dose and we can ignore the comment

(Addition I think we can use like react-query to fix fetch problem that you mention about pass user to be prop of every component )
![image](https://user-images.githubusercontent.com/88868804/221233209-1cde7a5a-c942-4f8c-9ce8-679f1821e176.png)

Reply:
@Nick-solich  I already fix 1) and 2) now ! you can see but the react-query. i will take some research and implement it in another branch ! thx !